lib/mapObjects/JsonRandom.cpp

```diff
@@ -87,7 +95,9 @@ namespace JsonRandom
 
 	TResources loadResource(const JsonNode & value, CRandomGenerator & rng)
 	{
-		std::string resourceName = loadKey(value, rng, "");
+		std::set<std::string> defaultResources(GameConstants::RESOURCE_NAMES, GameConstants::RESOURCE_NAMES + GameConstants::RESOURCE_QUANTITY - 2); //except mithril
```

Comment:
You can use std::begin / std::end instead of pointer arithmetics.
Also, I think this should be '-1', not '-2'. Or not?
```suggestion
		std::set<std::string> defaultResources(std::begin(GameConstants::RESOURCE_NAMES), std::end(GameConstants::RESOURCE_NAMES) - 1); //except mithril
```

Reply:
good catch, fixed